test/extended/util/framework.go

```diff
@@ -280,3 +281,19 @@ func ExtendedTestPath() string {
 func FixturePath(elem ...string) string {
 	return filepath.Join(append([]string{ExtendedTestPath()}, elem...)...)
 }
+
+// CheckURL grabs the output from the specified url and returns it.
+// It will retry up to N times if an error occurs during the request.
+func CheckURL(url string, retries int) (response string, err error) {
+	r, err := http.Get(url)
+	count := 0
+	for count < retries && err != nil {
```

Comment:
i like how "closures" suddenly makes it ok to follow all the horrible patterns we were taught to avoid w/ respect to global variables and threadlocals.